RestaurantManagementSystem/RestaurantManagementSystem/src/Core/Entities/RegistrationRequest.cs

```diff
@@ -46,24 +46,26 @@ public class RegistrationRequest
         }
         //create method
         public static RegistrationRequest Create(string name, string email
-            , string address, ContactInfo contact, RequestStatus.Status status, string verificationcode
+            , string address, string contactinfoemail, string contactinfophonenumber,Status status, string verificationcode
             )
         {
             Guid id = Guid.NewGuid();
             DateTime datecreated = DateTime.Now.ToUniversalTime();
             DateTime verificationvodeexpiresAt = datecreated.AddHours(24);
-            
+            ContactInfo contact = ContactInfo.Create(contactinfoemail, contactinfophonenumber);
             return new RegistrationRequest(id, name, email, address, contact, verificationcode, status, datecreated, verificationvodeexpiresAt);
         }
 
         //check if they are equal 
-        public static RegistrationRequest VertifyCode(string providedcode, RegistrationRequest obj)
+        
+        public  void VertifyCode(string providedcode)
         {
-            if(providedcode == obj.VerificationCode)
-            {
-                obj.Status = RequestStatus.Status.WaitingForAdminResponse;
+            if(this.Status ==Status.PendingVerification){
```

Comment:
there is not way to tell what happened when you call this method. this should either succeed or fail (throw errors)

```
public  void VertifyCode(string providedcode)
        {
            if(this.Status != Status.PendingVerification) {
                // throw domain error
            }
            if(this.VertifyCode != providedcode) {
                // throw domain error
            }
            this.Status = Status.PendingApproval
        }
        ```

Reply:
try to be explicit as much as you can